docs/DeveloperGuide.md

```diff
@@ -236,49 +299,127 @@ The following activity diagram summarizes what happens when a user executes a ne
 
 ### Confirmation + Clear command
+
 #### Steps to trigger
+
 1. User opens the app
 2. User enters `clear` (and subsequently sees a message asking to confirm)
 3. User enters `yes` to confirm the clear
+
 #### Implementation
+
 1. This features requires the state of the parser to be known.
-2. The parser is modified to store the previous taken in command, in this case whether the previous command was a successful clear command.
-3. If the previous command is not a clear command, it looks for the keyword clear. Otherwise, it looks for the keyword yes.
-4. Hence, the user will first need to call clear, before calling yes to invoke the clear mechanism, ensuring safety of data.
+2. The parser is modified to store the previous taken in command, in this case whether the previous command was a
+   successful clear command.
+3. If the previous command is not a clear command, it looks for the keyword clear. Otherwise, it looks for the keyword
+   yes.
+4. Hence, the user will first need to call clear, before calling yes to invoke the clear mechanism, ensuring safety of
+   data.
+
 #### Notes
-1. If you would like to extend the code for more features that require state, please do change the case condition for this feature.
-2. Currently, it follows the default commands if a word other than yes is given. But this will be improved in a future update.
+
+1. If you would like to extend the code for more features that require state, please do change the case condition for
+   this feature.
+2. Currently, it follows the default commands if a word other than yes is given. But this will be improved in a future
+   update.
 3. The state of the parser, rather than the app is used to reduce the chances of accidental clears.
 
+### Filter feature
+
+#### Implementation
+
+The filter feature works by updating the `Predicate` used in the `FilteredList<Applicant>` of `ModelManager`. Using
+the predicate, minimal changes to the implementation of StaffSnap is required.
+
+To create a single predicate that is able to search and filter for multiple fields, a `CustomFilterPredicate` class is
+created
+It currently contains the following fields and is able to filter for applicants which match all specified fields.
 
+1. Name
+2. Phone
+3. Email
+4. Position
+
+When `CustomFilterPredicate#test` is called, it will check if the specified fields are a substring of the same field of
+the applicant,
+returning true if all specified fields match, and false otherwise.
+
+#### Steps to trigger
+
+1. User opens the app
+2. User enters `filter [n/, e/, p/, hp/] [term]`, where one or more of the prefixes can be specified to be filtered by
+
+Once step 2 is complete, the GUI will update and refresh the applicant list with only applicants which match all
+specified fields.
+The following diagram summarises what happens when a user executes a Filter command:
+
+<puml src="diagrams/FilterCommandActivityDiagram.puml" alt="FilterCommandActivityDiagram" />
+
+### Design considerations
```

Comment:
Goodjob considering the possible alternatives and weighing the pros and cons of each alternative 💯